tests/functional/test_fusion_pg.py

```diff
@@ -0,0 +1,1639 @@
+    pg_mock.delete_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_api_init.return_value = pg_mock
+
+    op_mock = MagicMock()
+    op_mock.get_operation = MagicMock(side_effect=purefusion.rest.ApiException)
+    op_api_init.return_value = op_mock
+
+    with pytest.raises(AnsibleFailJson):
+        fusion_pg.main()
+
+    if pg_mock.get_placement_group.called:
+        pg_mock.get_placement_group.assert_called_with(
+            tenant_name="tenant1",
+            tenant_space_name="tenant_space1",
+            placement_group_name="placement_group1",
+        )
+    pg_mock.create_placement_group.assert_not_called()
+    pg_mock.update_placement_group.assert_not_called()
+    pg_mock.delete_placement_group.assert_not_called()
+    op_mock.get_operation.assert_not_called()
+
+
+@patch("fusion.OperationsApi")
+@patch("fusion.PlacementGroupsApi")
+def test_pg_create_ok(pg_api_init, op_api_init):
+    module_args = {
+        "name": "placement_group1",
+        "display_name": "some_display_name",
+        "tenant": "tenant1",
+        "tenant_space": "tenant_space1",
+        "region": "region1",
+        "availability_zone": "availability_zone1",
+        "storage_service": "storage_service1",
+        "state": "present",
+        "issuer_id": "ABCD1234",
+        "private_key_file": "private-key.pem",
+    }
+    set_module_args(module_args)
+
+    pg_mock = MagicMock()
+    pg_mock.get_placement_group = MagicMock(side_effect=purefusion.rest.ApiException)
+    pg_mock.create_placement_group = MagicMock(return_value=OperationMock("op1"))
+    pg_mock.update_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_mock.delete_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_api_init.return_value = pg_mock
+
+    op_mock = MagicMock()
+    op_mock.get_operation = MagicMock(return_value=OperationMock("op1", success=True))
+    op_api_init.return_value = op_mock
+
+    with pytest.raises(AnsibleExitJson) as excinfo:
+        fusion_pg.main()
+    assert excinfo.value.changed
+
+    pg_mock.get_placement_group.assert_called_with(
+        tenant_name="tenant1",
+        tenant_space_name="tenant_space1",
+        placement_group_name="placement_group1",
+    )
+    pg_mock.create_placement_group.assert_called_with(
+        purefusion.PlacementGroupPost(
+            name="placement_group1",
+            display_name="some_display_name",
+            availability_zone="availability_zone1",
+            region="region1",
+            storage_service="storage_service1",
+        ),
+        tenant_name="tenant1",
+        tenant_space_name="tenant_space1",
+    )
+    pg_mock.update_placement_group.assert_not_called()
+    pg_mock.delete_placement_group.assert_not_called()
+    op_mock.get_operation.assert_called_with("op1")
+
+
+@patch("fusion.OperationsApi")
+@patch("fusion.PlacementGroupsApi")
+def test_pg_create_without_display_name_ok(pg_api_init, op_api_init):
+    module_args = {
```

Comment:
You have here and below same module_args. To not just copy-pasting you can describe it as pytest.fixture and reuse this fixture everywhere, if you will need to delete or add values - just modify module_args dict

Reply:
Deduped common `state: present` and `state: absent` variants out for explicitness